online-shop/src/components/DetailsManagement.tsx

```diff
@@ -0,0 +1,297 @@
+  categories: string[];
+  reloadProductList: (data: IProduct[]) => void;
+  displayType: string;
+  displayProduct: (product: IProduct, msg: string) => void;
+  fetchUpdateProduct: (id: number, productUpdated: IProduct) => void;
+}
+class DetailsManagement extends React.Component<EditDetailsProps> {
+  categories: string[] = this.getCategories(this.props.productList);
+  updatedProduct: IProduct = {} as any;
+  productToDisplay: IProduct = {} as any;
+
+  defaultProduct: IProduct = {
+    id: 0,
+    name: "",
+    price: 0,
+    category: "",
+    description: "",
+    image: ""
+  };
+  onNameChange = (e: SyntheticEvent) => {
+    let nameUpdated: string;
+    nameUpdated = (e.target as HTMLInputElement).value.trim();
+    if (!/^\d+$/.test(nameUpdated)) {
+      this.productToDisplay.name = nameUpdated;
+    } else {
+      alert("Name is not corect!");
+    }
+  };
+
+  onPriceChange = (e: SyntheticEvent) => {
+    let price: any = "";
+    price = (e.target as HTMLInputElement).value.trim();
+    if (Number(price) && price > 0) {
+      this.productToDisplay.price = price;
+    } else {
+      alert("Price is not correct!");
+    }
+  };
+  onImageChange = (e: SyntheticEvent) => {
+    this.productToDisplay.image = (e.target as HTMLInputElement).value.trim();
+  };
+  onCategoryChange = (e: SyntheticEvent) => {
+    this.productToDisplay.category = (e.target as HTMLInputElement).value.trim();
+  };
+  onDetailsChange = (e: SyntheticEvent) => {
+    this.productToDisplay.description = (e.target as HTMLInputElement).value.trim();
+  };
+
+  createUpdatedProduct = (id: number): IProduct => {
+    return this.productToDisplay;
+  };
+  constructor(props: EditDetailsProps) {
+    super(props);
+    this.checkIfNewOrEdit();
+  }
+
+  checkIfNewOrEdit() {
+    if (this.props.displayType === "add") {
+      this.props.displayProduct(this.defaultProduct, "add");
+      this.render();
+    }
+  }
+  componentWillMount() {
+    this.checkIfNewOrEdit();
+  }
+  componentDidUpdate() {
+    this.checkIfNewOrEdit();
+  }
+
+  updateProduct = () => {
+    this.props.fetchUpdateProduct(this.props.product.id, this.productToDisplay);
+  };
+  render() {
+    let displayCategories = [...this.props.categories].map((category, key) => (
+      <option>{category}</option>
+    ));
+    let displayPage = () => {
+      if (this.props.displayType === "add") {
+        this.productToDisplay = this.defaultProduct;
+      } else {
+        this.productToDisplay = this.props.product;
+      }
+      return (
+        <div className="formItem">
+          <div className="field is-horizontal">
+            <div className="field-label is-normal">
+              <label className="label">Name</label>
+            </div>
+            <div className="field-body">
+              <div className="field">
+                <p className="control is-expanded has-icons-left">
+                  <input
+                    className="input is-primary"
```

Comment:
This component is huge - you should split it up in smaller components.

Reply:
Also, this component is neither presentational (dumb) nor container (smart). Split it up into smart / dumb components.